cdap-docs/developers-manual/source/building-blocks/applications.rst

```diff
@@ -48,9 +48,78 @@ Other components are defined using user-written classes that implement correspon
 interfaces and are referenced by passing an object, in addition to being assigned a unique
 name.
 
-Names used for streams and datasets need to be unique across the CDAP instance, while
+Names used for streams and datasets need to be unique across the CDAP namespace, while
 names used for programs and services need to be unique only to the application.
 
+Applications can use a ``Config`` class to receive a configuration during deployment time of the Application.
+For example, this can be used to specify the stream, dataset to create/read from at application deployment time
+instead of hard-coding them in the ``AbstractApplication``'s ``configure`` method. The configuration class needs to be
+the type parameter of the ``AbstractApplication`` class. It should also extend the ``Config`` class present in the CDAP API.
+The configuration is provided during application deployment time through a Header in the RESTful API. It is available during
+configuration time through the ``getConfig()`` method in ``AbstractApplication``.
+
+Information about the RESTful call is available in the :ref:`Lifecycle HTTP RESTful API documentation <http-restful-api-lifecycle>`.
+
+We can modify the ``MyApp`` class above to take in a Configuration ``MyApp.MyAppConfig``::
+
+      public class MyApp extends AbstractApplication<MyApp.MyAppConfig> {
```

Comment:
This code block can be moved over so that it is indented two spaces from the edge.